zerver/lib/message.py

```diff
@@ -634,13 +635,14 @@ def bulk_access_messages(user_profile: UserProfile, messages: Sequence[Message])
             filtered_messages.append(message)
     return filtered_messages
 
-def bulk_access_messages_expect_usermessage(
+def bulk_access_messages_expect_usermessage_including_deleted(
         user_profile_id: int, message_ids: Sequence[int]) -> List[int]:
     '''
     Like bulk_access_messages, but faster and potentially stricter.
 
```

Comment:
This docstring is no longer accurate after this change -- this function now behaves differently from `bulk_access_messages` so that it can return results that the latter function would not.

The purpose of this function, like its siblings `bulk_access_messages` and `access_message`, is to enforce privacy. For that reason it's important that it have a simple, clear interface that we can briefly and accurately explain. It might be possible to meet that standard with a change in this direction, but I think it'll be challenging, and this change doesn't.